var/spack/repos/builtin/packages/unzip/package.py

```diff
@@ -22,6 +22,7 @@ class Unzip(MakefilePackage):
     # clang and oneapi need this patch, likely others
     # There is no problem with it on gcc, so make it a catch all
     patch("configure-cflags.patch")
+    patch("strip.patch", when="%cce@:18.0.1")
```

Comment:
I'd normally suggest avoiding uploading patches and instead using configure arguments or a `filter_file` in `def patch(self)`, but in this case I think we do want to know when the patch stops to apply cleanly since we don't run CI on cce. For that reason, I think it might be useful to apply the patch always, not just on cce.

Reply:
I'm not well versed with filter_file() but I should be able to convert.

If we apply the patch always (all compiler/version) as you request, then how should I express that it should stop at cce%18.0.1 (which you seem to be in favor of).

Reply:
My understanding is that `-Wl,-s` is always correct (and arguably preferred when not calling `ld` directly), not just for `cce` but for all compilers, and not just for `cce%:18.0.1` but for all versions. Essentially, I'm suggesting to treat the patch the same way as the patch above it.